arch/x86/core/x86_mmu.c

```diff
@@ -21,20 +21,21 @@ MMU_BOOT_REGION((u32_t)&_image_rom_start, (u32_t)&_image_rom_size,
 #ifdef CONFIG_APPLICATION_MEMORY
 /* User threads by default can read/write app-level memory. */
 MMU_BOOT_REGION((u32_t)&__app_ram_start, (u32_t)&__app_ram_size,
-		MMU_ENTRY_WRITE | MMU_ENTRY_USER);
+		MMU_ENTRY_WRITE | MMU_ENTRY_USER | MMU_ENTRY_EXECUTE_DISABLE);
 #endif
 
 /* __kernel_ram_size includes all unused memory, which is used for heaps.
  * User threads cannot access this unless granted at runtime. This is done
  * automatically for stacks.
  */
 MMU_BOOT_REGION((u32_t)&__kernel_ram_start, (u32_t)&__kernel_ram_size,
-		MMU_ENTRY_WRITE | MMU_ENTRY_RUNTIME_USER);
+		MMU_ENTRY_WRITE |
+		MMU_ENTRY_RUNTIME_USER |
+		MMU_ENTRY_EXECUTE_DISABLE);
 
 
 void _x86_mmu_get_flags(void *addr, u32_t *pde_flags, u32_t *pte_flags)
```

Comment:
pde_flags and pte_flags need to be u64_t if PAE is turned on

Reply:
same as above